Index all password store meta urls on passff init

sunaurus opened the pull request:
There are many situations in which a single pass entry corresponds to multiple completely different web urls. One way of handling this is to list all the urls a password corresponds to inside the password entry using an `url: ` meta tag. However, this is currently not supported by passff (#374). 

In the linked issue, the possibility of using `pass grep` to find these meta urls was discussed. This would generally require decrypting the entire password store on every login, which might be acceptable for some password stores with fast hardware, but probably not a very good experience when using for example a smartcard to decrypt.

What I'm instead proposing in this PR is to only decrypt the entire store once on the initial load of the addon and to create a `Map` of all meta urls contained in the store. This index of meta urls would live in memory and would have a structure like:

```
{
  'services/amazon': ['amazon.com', 'amazon.co.uk', 'audible.com'],
  ...
  'fitness/impuls.ee': ['impuls.ee', 'arcticsport.ee']
}
```  

If this index exists, then the `hostMatchQuality` function can use it as an additional way of checking for matching entries. For example, my amazon entry is located at `services/amazon`, but it's still working on audible.com:

![Screenshot of passff](https://i.imgur.com/FcMrUVY.png)

The trade-off with this approach is that the list of pass entries and corresponding urls is now sitting decrypted in the user's browser memory, however, I believe this is an acceptable trade-off for a lot of users, and as long as this functionality is kept optional and explained next to the setting to turn it on, I think this is the best solution to #374.

This PR is not complete, I would still need to add the corresponding setting to toggle this functionality. I just wanted to submit it early to get some feedback. Does this approach seem reasonable?

This depends on passff/passff-host#38

tuxor1337 commented:
I didn't look at the code, but the idea sounds okay to me as long as it's an optional feature that's disabled by default.

sunaurus commented:
Great! I added a setting with a description of how it works into the preferences page (disabled by default). Sadly, I don't know German, French or Russian, so I can't translate it for the other messages.json files. Other than that, this PR should be ready for code review.

tuxor1337 commented:
Nice, but there is still one thing to worry about: The Firefox extension will be upgraded automatically. But the host app won't. Do you force users to upgrade their host app manually even if they don't need the new feature? Otherwise you have to deactivate this feature (and make note of it in the preferences) if the user doesn't have the required host app version...

sunaurus commented:
Will the next version of passff-host be 1.2.0? I added a note to the preferences page that the indexing option requires a passff-host >=1.2.0 and I also added a warning when running the `grepMetaUrls` command with a version lower than 1.2.0 (as was done with the `otp` command).
